chromadb/test/property/test_collections.py

```diff
@@ -136,6 +136,11 @@ def modify_coll(
             coll.metadata = new_metadata
 
         if new_name is not None:
+            if new_name in self.existing and new_name != coll.name:
```

Comment:
Is this case going to be exceptionally rare? In #343 I was trying to see if I could add a new rule to exercise this case. Or should we just add a unit test? Feels like the wrong tool for the job to rely on a rare generation.

Reply:
That's a good question. Perhaps a unit test to validate this specific invalid behavior is the way to go?

Reply:
I think we should just add that. Want me to add that here?

Reply:
Just pushed a unit test. Adding to `test_api.py` for now though it should definitely be split out later whenever we do that.